api/types/stats.go

```diff
@@ -89,10 +89,15 @@ type NetworkStats struct {
 
 // Stats is Ultimate struct aggregating all types of stats of one container
 type Stats struct {
-	Read        time.Time    `json:"read"`
-	Network     NetworkStats `json:"network,omitempty"`
-	PreCPUStats CPUStats     `json:"precpu_stats,omitempty"`
-	CPUStats    CPUStats     `json:"cpu_stats,omitempty"`
-	MemoryStats MemoryStats  `json:"memory_stats,omitempty"`
-	BlkioStats  BlkioStats   `json:"blkio_stats,omitempty"`
+	Read time.Time `json:"read"`
+
+	// Network is for fallback stats where API Version <= 1.21
+	Network NetworkStats `json:"network,omitempty"`
```

Comment:
We rather move deprecated fields to a separated struct. Look what we do with `ContainerJSON` in this same file. We have a `ContainerJSONBase` with the common fields and we compose structs for specific versions with it and the fields required by each version of the api.